core/vtk/ttkContourAroundPoint/ttkContourAroundPoint.cpp

```diff
@@ -19,116 +22,132 @@
-  ttk::Memory memUseObj;
-  _out = static_cast<vtkUnstructuredGrid *>(outputs[0]);
 
-#if 0
-  makeDummyOutput();
-#else
-  if(!preprocessDomain(inputs[0]))
-    return -1;
-  if(!preconditionConstraints(static_cast<vtkUnstructuredGrid *>(inputs[1]),
-                              static_cast<vtkUnstructuredGrid *>(inputs[2])))
-    return -1;
+using Class = ttkContourAroundPoint;
+
+//----------------------------------------------------------------------------//
+
+int Class::FillInputPortInformation(int port, vtkInformation *info) {
+  switch(port) {
+    case 0:
+      info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkDataSet");
+      return 1;
+    case 1:
+      info->Set(
+        vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkUnstructuredGrid");
+      return 1;
+    case 2:
+      info->Set(
+        vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkUnstructuredGrid");
+      return 1;
+  }
+  return 0;
+}
+
+int Class::FillOutputPortInformation(int port, vtkInformation *info) {
+  switch(port) {
+    case 0:
+      info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkUnstructuredGrid");
+      return 1;
+    case 1:
+      info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkUnstructuredGrid");
+      return 1;
+  }
+  return 0;
+}
+
+//----------------------------------------------------------------------------//
+
+int Class::RequestData(vtkInformation *request,
+                       vtkInformationVector **iVec,
+                       vtkInformationVector *oVec) {
+
+  _outFld = vtkUnstructuredGrid::GetData(oVec, 0);
+  _outPts = vtkUnstructuredGrid::GetData(oVec, 1);
+
+  if(!preprocessFld(vtkDataSet::GetData(iVec[0])))
+    return 0;
+  if(!preprocessPts(vtkUnstructuredGrid::GetData(iVec[1]),
+                    vtkUnstructuredGrid::GetData(iVec[2])))
+    return 0;
   if(!process())
-    return -1;
+    return 0;
   if(!postprocess())
-    return -1;
-#endif
+    return 0;
 
-  std::ostringstream memUseStream;
-  memUseStream << std::fixed << std::setprecision(3)
-               << memUseObj.getElapsedUsage() << " MB";
-  dMsg(cout, memUseStream.str(), memoryMsg);
-  return 0;
+  return 1;
 }
 
-//------------------------------------------------------------------------------------------------//
+//----------------------------------------------------------------------------//
 
-bool ttkContourAroundPoint::preprocessDomain(vtkDataSet *dataset) {
-  if(ui_scalars == "") {
-    vtkErrorMacro("A scalar variable needs to be defined on the Domain");
+bool Class::preprocessFld(vtkDataSet *dataset) {
+  ttk::Triangulation *triangulation = ttkAlgorithm::GetTriangulation(dataset);
+  if(!triangulation)
     return false;
-  }
 
-  ttk::Triangulation *triangulation
-    = ttkTriangulation::getTriangulation(dataset);
-  if(!triangulation) {
-    vtkErrorMacro("No ttk::Triangulation could be gained from the input "
-                  "field");
+  auto scalars = GetInputArrayToProcess(0, dataset);
```

Comment:
can you please PR ttk-data to update your state file?
see https://github.com/topology-tool-kit/ttk-data/commit/ba0daada042a2c78dddb0c8700db9639d2791162 for an example.